megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -1538,16 +1615,38 @@ FiringPlan guessFullFiringPlan(final Entity shooter,
                 continue;
             }
 
-            final WeaponFireInfo shoot = buildWeaponFireInfo(shooter,
-                                                             shooterState,
-                                                             target,
-                                                             targetState,
-                                                             weapon,
-                                                             game,
-                                                             true);
+            final WeaponType wtype = (WeaponType) weapon.getType();
 
-            if (0 < shoot.getProbabilityToHit()) {
-                myPlan.add(shoot);
+            WeaponFireInfo bestShoot = null;
+
+            // Energy / ammo-independent weapons
+            if (AmmoType.T_NA == wtype.getAmmoType()) {
+                bestShoot = buildWeaponFireInfo(shooter, target, weapon, null, game, false);
+            } else {
+                ArrayList<Mounted> ammos;
+                ammos = shooter.getAmmo(weapon);
+
+                WeaponFireInfo shoot;
+                for (Mounted ammo: ammos) {
```

Comment:
This looks like it could get expensive. Since we're not picking ammo type during the movement phase (and these "guess" methods are intended to be a cheap way to evaluate expected firepower for a move), is there really an ammo type that would heavily influence what move we make? I suppose for stuff like MML launchers it might make a difference, but in most cases it won't. Do we want to spend some time cutting down unnecessary computations here even at the expense of accuracy?

The reason I ask is that move path processing is very performance-sensitive, so any additional load here is multiplied many-fold, especially for faster units with more MP/jump MP.